### PR TITLE
refac: Predicate pushdown when reading charges

### PR DESCRIPTION
**Pull request by JesperGomes**

### CODE REVIEW

File: source/databricks/calculation_engine/package/calculation/preparation/transformations/charges_reader.py [hunk 1]
```diff
@@ -11,17 +11,45 @@
...
-    charge_prices_df = table_reader.read_charge_price_points()
+def read_charges(
+    table_reader: TableReader,
+    period_start_datetime: datetime,
+    period_end_datetime: datetime,
+) -> DataFrame:
+    charge_master_data_df = table_reader.read_charge_master_data_periods(
+        period_start_datetime, period_end_datetime
+    )
+    charge_master_data_df = clamp_period(
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/preparation/transformations/charges_reader.py:31]: clamp? How about this
```suggestion
    charge_master_data_df = filter_by_period(
```

File: source/databricks/calculation_engine/package/calculation_input/table_reader.py [hunk 2]
```diff
@@ -118,27 +118,56 @@ def read_time_series_points(
...
-        df = self._spark.read.format("delta").load(path)
+        df = (
+            self._spark.read.format("delta")
+            .load(path)
+            .where(col(Colname.from_date) < period_end_datetime)
+            .where(
+                col(Colname.to_date).isNull()
+                | (col(Colname.to_date) > period_start_datetime)
+            )
+        )
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation_input/table_reader.py:133]: Aren't you moving in the opposite direction of what we talked about today?
I believe that these filterings should be added in the PreparedDataReader.

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/calculation_input/table_reader.py]: this was changed before our meeting. I forgot to set the PR as draft

File: source/databricks/calculation_engine/package/calculation/preparation/transformations/charges_reader.py (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/calculation_engine/package/calculation/preparation/transformations/charges_reader.py:55]: it's not really a filter. It sets to/from date to the specified limits if they exceed.

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**